src/Data/BitSet/Word.hs

```diff
@@ -124,7 +124,7 @@ insert = GS.insert
 
 -- | /O(1)/. Delete an item from the bit set.
 delete :: Enum a => a -> BitSet a -> BitSet a
-delete = GS.delete
+delete x xs = GS.delete x xs
```

Comment:
Look at the generated core with -O2. All will be fine if WS.delete is fully
applied and inlines, but if it's partially applied and passed to a
higher-order function that doesn't inline, the Bits dictionary may be
passed at runtime (I believe). The extra parameters there are to fully
apply GS.delete in WS.delete, so GHC will inline it when compiling
WS.delete.
On Jun 15, 2016 2:45 PM, "Sergei Lebedev" <notifications@github.com> wrote:

> In src/Data/BitSet/Word.hs
> <https://github.com/lambda-llama/bitset/pull/22#discussion_r67221750>:
>
> > @@ -124,7 +124,7 @@ insert = GS.insert
> >
> >  -- | /O(1)/. Delete an item from the bit set.
> >  delete :: Enum a => a -> BitSet a -> BitSet a
> > -delete = GS.delete
> > +delete x xs = GS.delete x xs
>
> Hmm, could you explain why the arguments are necessary here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lambda-llama/bitset/pull/22/files/231962e0e6a63fe63e1952826f97ca529f20a4e5#r67221750>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABzi_QRX_uV6tfFowIlj45kg27ygZu9Cks5qMEg4gaJpZM4IyfKO>
> .
>